src/parachain/amm.ts

```diff
@@ -420,14 +423,62 @@ export class DefaultAMMAPI implements AMMAPI {
         return [...standardPools, ...stablePools];
     }
 
+    private async _swapThroughStandardPoolsOnly(
+        trade: Trade,
+        minimumAmountOut: MonetaryAmount<CurrencyExt>,
+        recipient: string,
+        deadline: number | string
+    ): Promise<void> {
+        const { amountIn, amountOutMin, path } = encodeSwapParamsForStandardPoolsOnly(
+            this.api,
+            trade,
+            minimumAmountOut
+        );
+        const swapExtrinsic = this.api.tx.zenlinkProtocol.swapExactAssetsForAssets(
+            amountIn,
+            amountOutMin,
+            path,
+            recipient,
+            deadline
+        );
+
+        await this.transactionAPI.sendLogged(swapExtrinsic, this.api.events.zenlinkProtocol.AssetSwap, true);
+    }
+
+    private async _swapThroughStandardAndStablePools(
+        trade: Trade,
+        minimumAmountOut: MonetaryAmount<CurrencyExt>,
+        recipient: string,
+        deadline: number | string
+    ): Promise<void> {
+        const { amountIn, amountOutMin, path } = encodeSwapParamsForStandardAndStablePools(
+            this.api,
+            trade,
+            minimumAmountOut
+        );
+        const swapExtrinsic = this.api.tx.zenlinkSwapRouter.swapExactTokenForTokensThroughStablePool(
+            amountIn,
+            amountOutMin,
+            path,
+            recipient,
+            deadline
+        );
+
+        await this.transactionAPI.sendLogged(swapExtrinsic, this.api.events.zenlinkStableAmm.CurrencyExchange, true);
+    }
+
     async swap(
         trade: Trade,
         minimumAmountOut: MonetaryAmount<CurrencyExt>,
-        recipient: AddressOrPair,
+        recipient: string,
```

Comment:
Does the recipient string represent an Account?
If so, I would prefer if it was typed more strongly.

Reply:
Yes, it represent an account, updated :+1: